CRM/Contact/Form/Search/Criteria.php

```diff
@@ -158,6 +158,13 @@ public static function basic(&$form) {
     );
 
     $componentModes = CRM_Contact_Form_Search::getModeSelect();
+    $enabledComponents = CRM_Core_Component::getEnabledComponents();
+
+    // unset disabled components that must should have been enabled
+    // to the option be viable
+    if (!array_key_exists('CiviMail', $enabledComponents)) {
+        unset($componentModes['8']);
```

Comment:
@hotwer Bernardo can you remove 2 spaces from the indent so its indented by 6 spaces not 8